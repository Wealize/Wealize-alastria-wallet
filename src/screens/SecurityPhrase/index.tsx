import React, { useState, useEffect } from 'react'
import { NavigationFunctionComponent } from 'react-native-navigation'
import Clipboard from '@react-native-community/clipboard'
import FlashMessage from 'react-native-flash-message'

import ViewContainer from '../../components/ViewContainer'
import { ALERT, SECURITY_PHRASE } from '../../constants/text'
import { getSecurityPhrase } from '../../utils/keychain'
import { Colors } from '../../utils/themes'
import {
  ContainerBottom,
  Subtitle,
  StyledButton,
  Title,
  BlurredText,
  BlurredPhrase,
  BlurredContainer,
  TouchableBlurred
} from './styles'
import { IMG } from '../../constants/urlImages'
import autologin from '../../utils/AutoLogin'
import { showSuccessNotification } from '../../utils/notifications'

const SecurityPhrase: NavigationFunctionComponent = (options) => {
  const [copiedPhrase, setCopiedPhrase] = useState('')
  const [showedPhrase, setShowedPhrase] = useState(false)

  const copyAndShow = () => {
    Clipboard.setString(copiedPhrase)
    setShowedPhrase(!showedPhrase)
  }

  const showPin = async () => {
    const pin = await getSecurityPhrase()
    setCopiedPhrase(pin)
  }

  const showFirstNotification = (): void => {
    const isFirstLogin = Object.values(options).includes(true)
    if (isFirstLogin) {
      showSuccessNotification(
        ALERT.CREATION_SUCCESS.TITLE,
        ALERT.CREATION_SUCCESS.SUBTITLE
      )
    }
  }

  useEffect(() => {
    showPin()
    showFirstNotification()
  }, [])

  return (
    <ViewContainer>
      <Title>{SECURITY_PHRASE.TITLE}</Title>
      <Subtitle>{SECURITY_PHRASE.SUBTITLE}</Subtitle>

      <TouchableBlurred onPress={copyAndShow}>
        <BlurredPhrase
          editable={false}
          multiline
          caretHidden
          value={copiedPhrase}
          onChangeText={(copiedPhrase) => setCopiedPhrase(copiedPhrase)}
        />
        {!showedPhrase && (
          <BlurredContainer source={IMG.SECURITY_PHRASE} style={{}}>
            <BlurredText>{SECURITY_PHRASE.HIDDEN_TITLE}</BlurredText>
          </BlurredContainer>
        )}
      </TouchableBlurred>

      <ContainerBottom>
        <StyledButton
          onPress={() => autologin()}
          title={SECURITY_PHRASE.BUTTON}
        />
      </ContainerBottom>
      <FlashMessage position="bottom" />
    </ViewContainer>
  )
}

SecurityPhrase.options = {
  statusBar: {
    backgroundColor: Colors.mainBackground,
    style: 'dark'
  },
  topBar: {
    visible: false,
    elevation: 0, // Remove border in Android
    borderColor: Colors.mainBackground // Remove border in iOS
  }
}

export default SecurityPhrase
