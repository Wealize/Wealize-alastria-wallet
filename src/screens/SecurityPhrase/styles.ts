import {
  Text,
  View,
  TextInput,
  ImageBackground,
  TouchableOpacity
} from 'react-native'
import { Button } from 'react-native-elements'
import styled from 'styled-components'

import { Colors } from '../../utils/themes'

export const ContainerBottom = styled(View)`
  flex: 1;
  justify-content: flex-end;
  align-items: center;
  margin: 20px auto 40px auto;
  width: 100%;
`

export const Title = styled(Text)`
  padding-top: 9%;
  margin-bottom: 18%;
  font-size: 24px;
  font-weight: bold;
`

export const Subtitle = styled(Text)`
  font-size: 14px;
  text-align: justify;
  margin-bottom: 12%;
`

export const TouchableBlurred = styled(TouchableOpacity)`
  height: 15%;
`

export const BlurredContainer = styled(ImageBackground).attrs({
  blurRadius: 10,
  imageStyle: {
    borderRadius: 20
  }
})`
  width: 100%;
  height: 100%;
  opacity: 0.7;
  position: absolute;
`

export const BlurredText = styled(Text)`
  position: absolute;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  text-align: center;
  top: 30%;
  font-size: 18px;
  color: rgba(0, 0, 0, 0.8);
  font-weight: bold;
`

export const BlurredPhrase = styled(TextInput)`
  height: 100%;
  border-radius: 20px;
  background-color: rgba(0, 0, 0, 0.05);
  text-align: center;
`

export const StyledButton = styled(Button).attrs({
  buttonStyle: {
    height: 39,
    width: 224,
    borderRadius: 45,
    backgroundColor: Colors.mainButton
  }
})``
