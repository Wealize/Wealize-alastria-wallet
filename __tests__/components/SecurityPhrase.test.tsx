import React from 'react'
import renderer from 'react-test-renderer'

import SecurityPhrase from '../../src/screens/SecurityPhrase'


describe('OnboardginComponent', () => {
  it('should render correctly', () => {
    const result = renderer.create(<SecurityPhrase componentId={'1'} />)
    expect(result).toMatchSnapshot()
  })
})