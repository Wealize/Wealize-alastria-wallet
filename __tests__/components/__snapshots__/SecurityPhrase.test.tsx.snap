// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardginComponent should render correctly 1`] = `
<View
  className="sc-bdvvtL jKiuHH"
>
  <Text
    className="sc-dkPtRN dldGBz"
  >
    Frase de seguridad
  </Text>
  <Text
    className="sc-hKwDye kWapWF"
  >
    Esta frase de seguridad permite recuperar tu cuenta en caso de que olvides el acceso. Recomendamos que la memorices o copies en un lugar seguro. Puedes consultarla siempre que lo necesites tras iniciar sesión. Por seguridad, nunca la reveles a otra persona.
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      caretHidden={true}
      className="sc-iCfMLu fNhBmV"
      editable={false}
      multiline={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={Object {}}
    >
      <Image
        blurRadius={10}
        className="sc-jRQBWg UfVrh"
        source={
          Object {
            "testUri": "../../../src/assets/img/background-grey.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            Object {
              "borderRadius": 20,
            },
          ]
        }
      />
      <Text
        className="sc-gKclnd hUXvTg"
      >
        Pulse aquí para revelar su frase de seguridad
      </Text>
    </View>
  </View>
  <View
    className="sc-gsDKAQ dDUTFT"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ca2b63",
              "borderColor": "#2089dc",
              "borderRadius": 45,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 39,
              "justifyContent": "center",
              "padding": 8,
              "width": 224,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Continuar
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 48,
          },
        ],
        "zIndex": 99,
      }
    }
  />
</View>
`;
